runner/ollamarunner/runner.go

```diff
@@ -591,18 +597,25 @@ func (s *Server) completion(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	var grammar *sample.Grammar
+	var err error
+	if req.Grammar != "" {
+		grammar, err = sample.NewGrammar(s.vocab, req.Grammar)
+		if err != nil {
+			http.Error(w, "failed to load model vocabulary required for format", http.StatusInternalServerError)
+			return
+		}
+	}
```

Comment:
I think it makes more sense for this to be in `NewSampler` so that runner can stay free of sampling setup related things

Reply:
I see what you mean, but in that case we'd have to pass `s.vocab` to `NewSampler` which would be a less clear of a parameter, vs passing an optional grammar to consider when sampling.

Reply:
Ah yeah it also doesn't make sense to have vocab loading code in the sampler. This is cool then 👍🏽